proposed/2022/vulnerabilities-in-restore.md

```diff
@@ -80,6 +80,14 @@ This feature will be opt-in to start and gather feedback from developers.
 
 To enable the feature, a developer can add `<NuGetAudit>enable</NuGetAudit>` to their project file as a MSBuild property. To disable the feature, a developer can add `<NuGetAudit>disable</NuGetAudit>` or remove the property from the project file.
 
+#### Setting Vulnerability Auditing Modes
+
+There will be different modes to audit vulnerabilities based on the developer's or developer's team preference. To do this, a developer will opt-in to a feature called `<NuGetAuditMode>` which will have different modes such as `direct`, `transitive`, and `all`.
+
+These modes should be pretty straight-forward. `direct` will scan for any top-level vulnerabilities, `transitive` will scan for any transitive-level vulnerabilities, and `all` will scan for both top-level and transitive-level vulnerabilities. The default will be `direct` until the experience is ready to be `all` given that transitive vulnerabilities are the majority of vulnerability notices (90%+).

```

Comment:
Yes let's modify this. Do we need a mode for `framework` or `sdk` dependencies? i.e. packages that are apart of a framework/sdk?

Reply:
Honestly, I don't know.

On one hand, implicit packages (those that MSBuild SDKs & Visual Studio project systems bring in) are intended not to be upgradable by customers. In VS, we prevent those packages from appearing in the "updates" tab, and on the browse & installed tab, up disable the update button. It's the SDK owner's responsibility to upgrade packages, which means customers will need to update to a newer version of the MSBuild SDK.

On the other hand, from a security perspective, the customer's app might use a package with a known vulerability. It might be difficult for them to figure out how to upgrade, and NuGet can't provide generic instructions on how to upgrade. Some SDKs provide their own extensibility (MSBuild property). Other times a `<PackageReference Update` (rather than `Include`) might work, but it depends on MSBuild import order. Other times using a different SDK version is needed, but that assumes that the SDK owner published a new version of their SDK that updated their implicit package versions. But at least by alerting the customer that their project uses a package with a known vulerability they have more knowledge and can investigate.

Honestly, my expectation is that implicit packages should mostly be "build related" packages, not packages with runtime assets, but of course there's nothing technically preventing SDK owners to add packages with runtime assets. So, I think implicit packages should mostly be a non-issue, but technically it could be.

This is potentially something else we can discuss in the meeting you're organizing, if we don't already have more topics than time available.

Reply:
Probably best to just skip this for now then until more feedback.